plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -2291,17 +2307,133 @@ else if (yaw < 315) {
             if (!attribute.hasParam()) {
                 return null;
             }
-            double angle = attribute.getDoubleParam();
-            double cos = Math.cos(angle);
-            double sin = Math.sin(angle);
-            double x = (object.getX() * cos) - (object.getY() * sin);
-            double y = (object.getX() * sin) + (object.getY() * cos);
+            double[] values = getRotatedAroundZ(attribute.getDoubleParam(), object.getX(), object.getY());
             Location location = object.clone();
-            location.setX(x);
-            location.setY(y);
+            location.setX(values[0]);
+            location.setY(values[1]);
             return new LocationTag(location);
         });
 
+        // <--[tag]
+        // @attribute <LocationTag.points_around_x[radius=<#.#>;points=<#>]>
+        // @returns ListTag(LocationTag)
+        // @group math
+        // @description
+        // Returns a list of points in a circular rotation around a location's x axis with the specified radius and number of points.
+        // For example: <player.location.points_around_x[radius=10;points=16]>
+        // -->
+        tagProcessor.registerTag(ListTag.class, "points_around_x", (attribute, object) -> {
+            if (!attribute.hasParam()) {
+                return null;
+            }
+            MapTag inputMap = attribute.paramAsType(MapTag.class);
+            if (inputMap == null) {
+                return null;
+            }
+            ObjectTag radiusObj = inputMap.getObject("radius");
+            ObjectTag pointsObj = inputMap.getObject("points");
+            if (radiusObj == null || pointsObj == null) {
+                return null;
+            }
+            ElementTag radius = radiusObj.asType(ElementTag.class, attribute.context);
+            ElementTag amount = pointsObj.asType(ElementTag.class, attribute.context);
+            if (radius == null || amount == null) {
+                return null;
+            }
+            if (amount.asInt() < 1) {
+                attribute.echoError("Invalid amount of points! There must be at least 1 point.");
+            }
+            double angle = 2 * Math.PI / amount.asInt();
+            ListTag points = new ListTag();
+            for (int i = 1; i <= amount.asInt(); i++) {
+                double[] result = getRotatedAroundX(angle * i, radius.asDouble(), 0);
+                LocationTag newLocation = object.clone();
+                newLocation.add(0, result[0], result[1]);
```

Comment:
Combine the above two lines into one... never type `somevec.add(x,y,z);` on its own line (or any similar method) as that, uh... technically makes sense to do under Bukkit's vector API but doesn't make sense in any others, and trusting Bukkit's semantics is a bad habit to build